force-app/main/default/classes/DMLOperations.cls

```diff
@@ -181,14 +229,16 @@ public with sharing class DMLOperations {
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
 		// Loop through each Contact
-
-		// Extract the Account Name from Contact's LastName
-
-		// Get the account based on the name or create one if it doesn't exist
-
-		// Link the Contact to its Account using the Account Id
-
+        for (Contact contact : contactsToUpsert) {
+            // Extract the Account Name from Contact's LastName
+            String accountName = contact.LastName;
+            // Get the account based on the name or create one if it doesn't exist
+            Account acc = upsertAccount(accountName);
```

Comment:
I know the directions say you can do this, but can you refactor the code to not do DML within the loop? Maybe look at refactoring the method for question 9 to bulkify it?